packages/jest-runtime/src/index.ts

```diff
@@ -932,6 +1033,24 @@ class Runtime {
     return require(moduleName);
   }
 
+  private _importCoreModule(moduleName: string, context: VMContext) {
+    const required = this._requireCoreModule(moduleName);
+
+    return new SyntheticModule(
+      ['default', ...Object.keys(required)],
+      function () {
+        // @ts-ignore: TS doesn't know what `this` is
+        this.setExport('default', required);
+        Object.entries(required).forEach(([key, value]) => {
+          // @ts-ignore: TS doesn't know what `this` is
+          this.setExport(key, value);
+        });
+      },
+      // should identifier be `node://${moduleName}`?
+      {context, identifier: moduleName},
```

Comment:
Normally stack traces will show e.g. `fs.js` as the identifier. Have you tried a `fs.readFileSync('/does/not/exist')` to see what shows up with this logic?

Reply:
Doesn't seem to show up at all

```sh-session
$ node --experimental-vm-modules file.mjs
(node:93098) ExperimentalWarning: The ESM module loader is experimental.
Error: ENOENT: no such file or directory, open '/does/not/exist'
    at Object.openSync (fs.js:462:3)
    at Module.readFileSync (fs.js:364:35)
    at myTestingModule.js:3:6
    at SourceTextModule.evaluate (internal/vm/module.js:221:32)
    at file:///Users/simen/repos/jest/file.mjs:35:22 {
  errno: -2,
  syscall: 'open',
  code: 'ENOENT',
  path: '/does/not/exist'
}
(node:93098) ExperimentalWarning: VM Modules is an experimental feature. This feature could change at any time
```

Test:

```js
import {SourceTextModule, SyntheticModule} from 'vm';
import {createRequire} from 'module';

const require = createRequire(import.meta.url);
const fs = require('fs');

const required = fs;

const module = new SourceTextModule(
  `
  import * as fs from 'fs';
  fs.readFileSync('/does/not/exist');
`,
  {identifier: 'myTestingModule.js'},
);

module
  .link(async () => {
    const m = new SyntheticModule(
      ['default', ...Object.keys(required)],
      function () {
        this.setExport('default', required);
        Object.entries(required).forEach(([key, value]) => {
          this.setExport(key, value);
        });
      },
      {identifier: 'fs halla'},
    );

    await m.link(() => {});
    await m.evaluate();

    return m;
  })
  .then(() => module.evaluate())
  .catch(error => {
    process.exitCode = 1;

    console.error(error);
  });
```